keps/sig-cli/0031-kustomize-integration.md

```diff
@@ -167,30 +256,151 @@ To get the detailed objects in a kustomization
 kubectl get -f <dir> --dry-run -o yaml|json
 ```
 
-#### delete
+##### delete
 The command visible to users is
+
 ```
 kubectl delete -f <dir>
 ```
 
-### Implementation Details/Notes/Constraints
+#### Why we like this approach
+
+- It is capable of friction that requires deeper integration - such as producing errors referencing line
+  numbers of the original files (rather than the output files).
+- It integrates will all commands - get, delete, etc - would automatically work.
+- It is more consistent with UX workflow for commands that work directly off a target
+- It has a cleaner and simpler UX than Option A, fewer steps
+- It integrates the solution directly into the commands with issues instead of tacking it on as separate step
+- Users less likely to get confused and accidentally apply a directory with a kustomization.yaml and have
+  it do the wrong thing (it will do the right thing).
+
+#### Why don't like this approach
+
+- Kustomize does more than the resource builder currently does and this could surprise users
+- Commands are doing more than they were before
```

Comment:
Not backwards compatible, which is a big deal breaker!

Reply:
Is it not?  The error behavior is a bit different, but you would get an error before.

Reply:
The one where it picks `kustomize.yaml` based on name is breaking backwards compatibility. Because beforehand nothing as such would be picked or iirc would fail. With kustomize in it will pick it up and proceed with transformations, that's not backwards compatible. With the option where you explicitly point to `kustomize.yaml` that is fine.

Reply:
If it used to fail doesn't anymore does this break backwards compatibility?  If it used to create some resources but fail on the kustomize resource and then exit non-0 is that breaking backwards compatibility?  Guarantees do we provide in the cases where a command fails, are the documented?

Reply:
Addressing this by pulling into another KEP